### PR TITLE
use local function to prepend unique value

### PR DESCRIPTION
**Pull request by dirk-thomas**
Needed for colcon/colcon-core#216.

### CODE REVIEW

File: connext_cmake_module/env_hook/connext.sh.in [hunk 1]
```diff
@@ -33,7 +33,58 @@ else
...
-ament_prepend_unique_value PATH "$_NDDSHOME_TO_USE/bin"
+# function to prepend non-duplicate values to environment variables
+# using colons as separators and avoiding trailing separators
+_connext_prepend_unique_value() {
+  # arguments
+  _listname="$1"
+  _value="$2"
+  #echo "listname $_listname"
+  #eval echo "list value \$$_listname"
+  #echo "value $_value"
```
Comment (jacobperron) [inline connext_cmake_module/env_hook/connext.sh.in:46]: Remove these commented statements, unless there's something left TODO.

File: connext_cmake_module/env_hook/connext.sh.in [hunk 2]
```diff
@@ -33,7 +33,58 @@ else
...
+    fi
+  done
+  unset _item
+
+  # prepend only non-duplicates
+  if [ -z "$_duplicate" ]; then
+    # avoid trailing separator
+    if [ -z "$_values" ]; then
+      eval export $_listname=\"$_value\"
+      #eval echo "set list \$$_listname"
```
Comment (jacobperron) [inline connext_cmake_module/env_hook/connext.sh.in:72]: remove

File: connext_cmake_module/env_hook/connext.sh.in [hunk 3]
```diff
@@ -33,7 +33,58 @@ else
...
+    else
+      # field separator must not be a colon
+      unset IFS
+      eval export $_listname=\"$_value:\$$_listname\"
+      #eval echo "prepend list \$$_listname"
```
Comment (jacobperron) [inline connext_cmake_module/env_hook/connext.sh.in:77]: remove

File: connext_cmake_module/env_hook/connext.sh.in [hunk 4]
```diff
@@ -33,7 +33,58 @@ else
...
+  _value="$2"
+  #echo "listname $_listname"
+  #eval echo "list value \$$_listname"
+  #echo "value $_value"
+
+  # check if the list contains the value
+  eval _values=\"\$$_listname\"
+  _duplicate=
+  _connext_prepend_unique_value_IFS=$IFS
+  IFS=":"
```
Comment (jacobperron) [inline connext_cmake_module/env_hook/connext.sh.in:52]: It doesn't appear that `IFS` is used meaningfully in this function other than being set at the end (L78). Is something missing or can this statement be removed?

File: connext_cmake_module/env_hook/connext.sh.in (hunk 1 above)
Comment (dirk-thomas) [reply connext_cmake_module/env_hook/connext.sh.in:46]: The function is a copy from here: https://github.com/ament/ament_package/blob/6c8acd25498d2674e7ad9d316b35d112ae1ab8c1/ament_package/template/package_level/local_setup.sh.in#L45-L94

The comments are there to easily comment them in for debugging if needed. So I would rather keep them.

File: connext_cmake_module/env_hook/connext.sh.in (hunk 2 above)
Comment (dirk-thomas) [reply connext_cmake_module/env_hook/connext.sh.in:72]: See other comment.

File: connext_cmake_module/env_hook/connext.sh.in (hunk 3 above)
Comment (dirk-thomas) [reply connext_cmake_module/env_hook/connext.sh.in:77]: See other comment.

File: connext_cmake_module/env_hook/connext.sh.in (hunk 1 above)
Comment (jacobperron) [reply connext_cmake_module/env_hook/connext.sh.in:46]: Okay. Might be worth a comment indicating where this was copied from.

File: connext_cmake_module/env_hook/connext.sh.in (hunk 4 above)
Comment (dirk-thomas) [reply connext_cmake_module/env_hook/connext.sh.in:52]: Actual it is being used by the following `for` loop. `_values` is being iterated by considering the `IFS` to determine how to split the string.

Comment (jacobperron) [reply connext_cmake_module/env_hook/connext.sh.in:52]: ah I see. I didn't release that was it was used for. TIL.

**Review by jacobperron — APPROVED** [review]

File: connext_cmake_module/env_hook/connext.sh.in (hunk 1 above)
Comment (dirk-thomas) [reply connext_cmake_module/env_hook/connext.sh.in:46]: Good point: fc829716eb8a2a1c5cb06b5e00c61ed5e3f55c0a.

**Review by jacobperron — APPROVED** [review]

**Pull request merged**